src/vm/analysis/type_checker/natives/assets.rs

```diff
@@ -186,6 +190,41 @@ pub fn check_special_transfer_token(
     )
 }
 
+pub fn check_special_stx_transfer(
+    checker: &mut TypeChecker,
+    args: &[SymbolicExpression],
+    context: &TypingContext,
+) -> TypeResult {
+    let memo_passed;
+    if let Ok(()) = check_argument_count(4, args) {
+        memo_passed = true;
+    } else {
+        check_argument_count(3, args)?;
+        memo_passed = false;
+    }
```

Comment:
Style nit: rust's treatment of code blocks lets write these kinds of things without separating the declaration from the assignment:

```suggestion
    let memo_passed = let Ok(()) = check_argument_count(4, args) {
        true
    } else {
        check_argument_count(3, args)?;
        false
    };
```
```